iron-doc-viewer.html

```diff
@@ -85,30 +107,78 @@
       Polymer({
         is: 'iron-doc-viewer',
 
-        behaviors: [Polymer.IronDocBehavior],
+        behaviors: [Polymer.IronDocViewerBehavior],
 
         properties: {
-          rootNamespace: {
+          /**
+           * Path to the item in the descriptor to display.
+           *
+           * Examples:
+           *   - `/` for the root namespace
+           *   - `/elements/paper-button`
+           *   - `/mixins/paper-button-behavior`
+           *
+           * Defaults to the current `window.location` path.
+           *
+           * If `baseHref` is set, it will be trimmed from the prefix of this
+           * `path`.
+           */
+          path: {
             type: String,
+            value: null
           },
 
-          _currentDescriptor: {
-            type: Object,
+          /**
+           * Instead of displaying items relative to the top level of
+           * `descriptor`, start from this namespace.
+           */
+          rootNamespace: {
+            type: String,
           },
 
           _descriptorType: String,
+
+          _pathSetExternally: {
+            type: Boolean,
+            value: false
+          },
+
+          _ourPath: String,
+
+          _ourPathChanging: Boolean,
+
+          _currentDescriptor: Object,
         },
 
-        observers: ['_dataChanged(descriptor,route.path)'],
+        observers: [
+          '_pathChanged(path)',
+          '_ourPathChanged(_ourPath)',
+          '_dataChanged(descriptor, path)'
+        ],
 
         _equal: function(a, b) {
           return a == b;
         },
 
-        _dataChanged: function(descriptor, routePath) {
+        _pathChanged: function(path) {
+          this._pathSetExternally = path && !this._ourPathChanging;
+        },
+
+        _ourPathChanged: function(ourPath) {
+          if (ourPath) {
+            this._ourPathChanging = true;
+            this.path = ourPath;
```

Comment:
Done. I may actually rip this feature out and require something more explicit (caller just provides the route directly or something).